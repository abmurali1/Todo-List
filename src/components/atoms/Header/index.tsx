import React, {FC, useEffect} from 'react';
import {StyleSheet, Text, Image, View, TouchableOpacity} from 'react-native';

interface HeaderProps {
  headerTitle: string;
  onPress: () => void;
}
const Header: FC<HeaderProps> = props => {
  return (
    <View style={styles.headerStyle}>
      <TouchableOpacity onPress={props.onPress}>
        <Image
          source={require('../../../assets/images/arrowRight/arrow-right.png')}
          style={styles.imageStyle}
        />
      </TouchableOpacity>
      <Text style={styles.textStyle}>{props.headerTitle}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  headerStyle: {
    flexDirection: 'row',
    borderBottomWidth: 0.2,
    borderBottomColor: 'black',
    width: '100%',
    justifyContent: 'flex-start',
    padding: 10,
    alignItems: 'center',
  },
  textStyle: {
    textAlign: 'center',
    fontSize: 20,
    fontFamily: 'Verdana',
    fontWeight: 'bold',
    color: '#1b1f16',
  },
  imageStyle: {
    height: 20,
    width: 20,
    transform: [{rotate: '180deg'}],
    marginRight: 100,
  },
});

export default Header;
