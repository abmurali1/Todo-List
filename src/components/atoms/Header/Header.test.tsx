import React from 'react';
import renderer from 'react-test-renderer';
import Header from './index';

describe('<Header>', () => {
  it('renders correctly', () => {
    const mockFn = jest.fn();
    const tree = renderer
      .create(<Header headerTitle="Click" onPress={mockFn} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
