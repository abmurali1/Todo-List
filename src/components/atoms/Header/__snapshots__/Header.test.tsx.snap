// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "black",
      "borderBottomWidth": 0.2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/arrowRight/arrow-right.png",
        }
      }
      style={
        Object {
          "height": 20,
          "marginRight": 100,
          "transform": Array [
            Object {
              "rotate": "180deg",
            },
          ],
          "width": 20,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#1b1f16",
        "fontFamily": "Verdana",
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Click
  </Text>
</View>
`;
